### PR TITLE
feature: auto trace SPA performance when router change

### PR DESCRIPTION
**Pull request by sosohime**
patch history.pushState to support SPA history change

### CODE REVIEW

**Comment by Fine0830** [comment]
I tested positive for COVID and have a high fever, let me review it after I recover.

File: src/monitor.ts [hunk 1]
```diff
@@ -42,9 +42,7 @@ const ClientMonitor = {
     };
     this.validateOptions();
     this.catchErrors(this.customOptions);
-    if (!this.customOptions.enableSPA) {
```
Comment (Fine0830) [inline src/monitor.ts:45]: Since the `enableSPA` turns on single page mode, you can't change this.

**Review by Fine0830 — COMMENTED** [review]
In this way, parameters(serviceVersion, pagePath...) cannot be changed according to different routes, so we use manual methods with `setPerformance`.

**Comment by sosohime** [comment]
> In this way, parameters(serviceVersion, pagePath...) cannot be changed according to different routes, so we use manual methods with `setPerformance`.

Okey，I understand.

It's still some question on these follow code, when `enableSPA` turns on, will add `hashchange` listener to report perfmence when app hashHistory changed. But `hashchange event` only triggered by set `location.hash = 'xxx'`, If app use history api(like `React-Router`, `Vue-Router`), it's not worked. We still need to patch `window.history.pushState`
[`https://github.com/apache/skywalking-client-js/blob/93c81dc8ab0cdb9591fde5caff51faeb48a3108e/src/performance/index.ts#L33-L40`](https://github.com/apache/skywalking-client-js/blob/93c81dc8ab0cdb9591fde5caff51faeb48a3108e/src/performance/index.ts#L31-L40)

**Comment by Fine0830** [comment]
Please read https://github.com/apache/skywalking-client-js#spa-page, and use manual reporting.

**Pull request closed without merging**